robottelo/manifests.py

```diff
@@ -17,160 +12,103 @@
 from robottelo.config import settings
 
 
-def get_tempfile():
-    """Creates a temporary file.
-
-    Generates temporary file with mkstemp,
-    closes the handle, and returns the path to temporary file.
-
+class ManifestCloner(object):
+    """Manifest clonning utility class."""
+    def __init__(self, template=None, signing_key=None):
+        self.template = template
+        self.signing_key = signing_key
+
+    def _download_manifest_info(self):
+        """Download and cache the manifest information."""
+        self.template = requests.get(settings.fake_manifest.url).content
+        self.signing_key = requests.get(settings.fake_manifest.key_url).content
+        self.signature = PKCS1_v1_5.new(RSA.importKey(self.signing_key))
+
+    def clone(self):
+        """Clones a RedHat-manifest file.
+
+        Change the consumer ``uuid`` and sign the new manifest with
+        signing key. The certificate for the key must be installed on the
+        candlepin server in order to accept uploading the cloned
+        manifest.
+
+        :return: A file-like object (``BytesIO`` on Python 3 and
+            ``StringIO`` on Python 2) with the contents of the cloned
+            manifest.
+        """
+        if self.signing_key is None or self.template is None:
+            self._download_manifest_info()
+
+        template_zip = zipfile.ZipFile(six.BytesIO(self.template))
+        consumer_export_zip = zipfile.ZipFile(
```

Comment:
Will do, but the code already says that since "some_zip.read == extract the file and return its contents". The usage of BytesIO throughout this code is just to avoid file creation.